server/src/test/java/org/elasticsearch/transport/ProxyConnectionStrategyTests.java

```diff
@@ -79,10 +83,21 @@ public void testProxyStrategyWillOpenExpectedNumberOfConnectionsToAddress() {
         try (MockTransportService transport1 = startTransport("node1", Version.CURRENT)) {
             TransportAddress address1 = transport1.boundAddress().publishAddress();
 
-            try (MockTransportService localService = MockTransportService.createNewService(Settings.EMPTY, Version.CURRENT, threadPool)) {
+            try (
+                MockTransportService localService = spy(MockTransportService.createNewService(Settings.EMPTY, Version.CURRENT, threadPool))
+            ) {
                 localService.start();
                 localService.acceptIncomingRequests();
 
+                // Handshake (as part of cluster name validation) should go through the internal remote connection
+                // So it can be intercepted accordingly
+                doAnswer(invocation -> {
+                    final Object connection = invocation.getArgument(0);
+                    assertThat(connection, isA(RemoteConnectionManager.InternalRemoteConnection.class));
```

Comment:
Could also include `assertThat(RemoteConnectionManager.resolveRemoteClusterAlias((Transport.Connection) connection).get(), equalTo(clusterAlias));` since that's functionally what we really care about.

Reply:
Good point. I updated to assert the cluster alias.